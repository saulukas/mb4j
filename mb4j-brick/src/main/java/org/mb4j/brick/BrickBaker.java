package org.mb4j.brick;

public interface BrickBaker<P> {
  Brick bakeBrick(P params);
}
