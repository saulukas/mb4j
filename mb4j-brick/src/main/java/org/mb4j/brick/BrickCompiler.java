package org.mb4j.brick;

public class BrickCompiler {

    public static <T extends MustacheBrick> BrickTemplate<T> compile(Class<T> brick) {
        return null;
    }

}
