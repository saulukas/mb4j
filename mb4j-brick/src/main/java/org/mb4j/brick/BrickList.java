package org.mb4j.brick;

import java.io.Writer;
import org.mb4j.brick.jmustache.Template.Fragment;
import org.mb4j.brick.renderer.RenderingScope;

public class BrickList<T extends MustacheBrick> extends MustacheBrick {

    public final Iterable<T> items;

    public BrickList(Iterable<T> items) {
        this.items = items;
    }

    @Override
    public void execute(Fragment frag, Writer out) {
        ((RenderingScope) out).renderList(items, frag);
    }
}
