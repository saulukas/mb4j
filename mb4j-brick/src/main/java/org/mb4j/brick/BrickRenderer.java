package org.mb4j.brick;

import org.mb4j.brick.renderer.RenderingScope;
import org.mb4j.brick.template.TemplateProvider;
import java.io.Writer;

public class BrickRenderer {
  private final TemplateProvider templateProvider;

  public BrickRenderer(TemplateProvider templateProvider) {
    this.templateProvider = templateProvider;
  }

  public void render(Brick brick, Writer out) {
    RenderingScope scope = new RenderingScope(out, templateProvider);
    scope.render(brick);
  }
}
