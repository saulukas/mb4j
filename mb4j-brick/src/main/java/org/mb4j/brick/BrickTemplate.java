package org.mb4j.brick;

public class BrickTemplate<T extends MustacheBrick> {

    static class Segment {

    }
}
