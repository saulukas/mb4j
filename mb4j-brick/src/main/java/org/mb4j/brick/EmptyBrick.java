package org.mb4j.brick;

import com.samskivert.mustache.Template;
import java.io.Writer;

public class EmptyBrick extends MustacheBrick {

    private static final EmptyBrick INSTANCE = new EmptyBrick();

    public static EmptyBrick emptyBrick() {
        return INSTANCE;
    }

    @Override
    public void execute(Template.Fragment frag, Writer out) {
    }
}
