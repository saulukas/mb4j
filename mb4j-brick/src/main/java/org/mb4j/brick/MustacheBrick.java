package org.mb4j.brick;

import com.samskivert.mustache.Mustache.Lambda;
import com.samskivert.mustache.Template.Fragment;
import java.io.Writer;
import org.mb4j.brick.renderer.RenderingScope;

public class MustacheBrick implements Lambda {

    @Override
    public void execute(Fragment frag, Writer out) {
        ((RenderingScope) out).renderIndented(this, frag);
    }
}
