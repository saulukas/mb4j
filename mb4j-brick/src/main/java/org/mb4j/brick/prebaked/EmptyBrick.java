package org.mb4j.brick.prebaked;

import com.samskivert.mustache.Template;
import java.io.Writer;
import org.mb4j.brick.Brick;

public class EmptyBrick extends Brick {
  private static final EmptyBrick INSTANCE = new EmptyBrick();

  public static EmptyBrick emptyBrick() {
    return INSTANCE;
  }

  @Override
  public void execute(Template.Fragment frag, Writer out) {
  }
}
