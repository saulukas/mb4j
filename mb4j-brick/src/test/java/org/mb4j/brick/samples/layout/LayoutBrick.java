package org.mb4j.brick.samples.layout;

import org.mb4j.brick.Brick;
import org.mb4j.brick.EmptyBrick;
import org.mb4j.brick.template.TemplateType;

@TemplateType(".mustache")
public class LayoutBrick extends Brick {
  Brick header = new HeaderBrick();
  Brick footer = new FooterBrick();
  Brick content = new EmptyBrick();
}
