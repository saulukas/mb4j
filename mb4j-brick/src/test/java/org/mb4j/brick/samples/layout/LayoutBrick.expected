
   Header: Enterprise System

   

   Footer: Â© 2013
